src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -480,11 +480,19 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
     }
     def fromHeader(name: ToolName, header: List[String]) = {
       import scala.sys.process.Parser.tokenize
-      val tag  = s"$name:"
-      val endc = "*" + "/"    // be forgiving of /* scalac: ... */
-      def stripped(s: String) = s.substring(s.indexOf(tag) + tag.length).stripSuffix(endc)
-      val lines = header.filter(_.contains(tag)).map(line => stripped(line).trim())
-      if (name == ToolName.filter) lines else lines.flatMap(tokenize)
+      val namePattern = raw"\s*//\s*$name:\s*(.*)".r
+      val optionsPattern = raw"\s*//>\s*using\s+option(s)?\s+(.*)".r
+      def matchLine(line: String): List[String] =
+        line match {
+          case namePattern(rest) => if (name == ToolName.filter) List(rest) else tokenize(rest)
+          case _ if name == ToolName.scalac =>
+            line match {
+              case optionsPattern(plural, rest) => if (plural == null) List(rest) else rest.split(',').toList.map(_.trim)
```

Comment:
maybe for trailing whitespace?

```suggestion
              case optionsPattern(plural, rest) => if (plural == null) List(rest.trim) else rest.split(',').toList.map(_.trim)
```

Reply:
Now that you mention it, one reason to prefer `// scalac: args` is that it uses `tokenize`. `//> using options x, y, z` is actually unfortunate. `//> using options -Wconf:cat=deprecation:s,cat=feature:e` what does it do?

I will trim for now and explore the complications later.

I was glad to use one standard for `tokenize` everywhere, but like the xkcd, let's standardize on scala-cli and now we have two standards.

Reply:
I guess they support double-quotes. Maybe sufficient to tokenize then strip trailing comma, also simple comma.
```
tokenize("""-Xlint, "-Wconf:cat=deprecation:s,cat=feature:e", -feature , -deprecation""")
```

Reply:
Not trimming was cut/paste from `filter`, which probably should trim and not take the raw line. The filter would inadvertently not match with spurious spaces.